src/lucky_record/errors.cr

```diff
@@ -0,0 +1,20 @@
+# = Lucky Record Errors
+#
+# Generic Lucky Record exception class.
+class LuckyRecordError < Exception
+end
+
+# Raised when Lucky Record cannot find a record by given id
+class RecordNotFound < LuckyRecordError
+  getter model : (String|Nil)
+  getter primary_key : (String|Nil)
+  getter id : (String|Int32|Nil)
+
+  def initialize(message : (String|Nil) = nil, model : (String|Nil) = nil, primary_key : (String|Nil) = nil, id : (String|Int32|Nil) = nil)
```

Comment:
In Crystal you can set the instance variable right away like this: `def initialize(@model, @id)`. You also don't need to redeclare the types since you did that in the getter